os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -431,6 +435,16 @@ void up_assert(const uint8_t *filename, int lineno)
 
 	board_led_on(LED_ASSERTION);
 
+#ifdef CONFIG_CPP_HAVE_VARARGS
+	va_list valist;
+	const char *assertInfo;
+
+	/* Extract the extra debug arguments if any */
+	va_start(valist, lineno);
+	assertInfo = va_arg(valist, const char *);
```

Comment:
defining ```assertInfo``` as ```const``` and then re assigning it might not work. I feel like its better to assign it during declaration.